cmd/xl-storage-disk-id-check.go

```diff
@@ -821,6 +825,61 @@ func (p *xlStorageDiskIDCheck) monitorDiskStatus() {
 	}
 }
 
+// monitorDiskStatus should be called once when a drive has been marked offline.
+// Once the disk has been deemed ok, it will return to online status.
+func (p *xlStorageDiskIDCheck) monitorDiskWritable() {
+	// We check every 2 minutes if the disk is writable and we can read back.
+	t := time.NewTicker(2 * time.Minute)
+	defer t.Stop()
+	fn := mustGetUUID()
+
+	// Be just above directio size.
+	toWrite := []byte{xioutil.DirectioAlignSize + 1: 42}
+	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
+	for range t.C {
+		if atomic.LoadInt32(&p.health.status) != diskHealthOK {
+			continue
+		}
+		goOffline := func(err error) {
+			if atomic.CompareAndSwapInt32(&p.health.status, diskHealthOK, diskHealthFaulty) {
+				logger.LogAlwaysIf(context.Background(), fmt.Errorf("node(%s): taking drive %s offline: %v", globalLocalNodeName, p.storage.String(), err))
+				go p.monitorDiskStatus()
```

Comment:
I think this is causing some kind of stack overflow on windows @klauspost

Reply:
Nah `atomic.CompareAndSwapInt32` will only trigger once. It is not recursive in case that is what you are thinking.